roofit/roofitcore/src/RooTreeDataStore.cxx

```diff
@@ -369,7 +369,7 @@ void RooTreeDataStore::loadValues(const TTree *t, const RooFormulaVar* select, c
   // We need a custom deleter, because if we don't deregister the Tree from the directory
   // of the original, it tears it down at destruction time!
   auto deleter = [](TTree* tree){tree->SetDirectory(nullptr); delete tree;};
-  std::unique_ptr<TTree, decltype(deleter)> tClone(static_cast<TTree*>(t->Clone()), deleter);
+  std::unique_ptr<TTree, decltype(deleter)> tClone{static_cast<TTree*>(t->GetDirectory()->CloneObject(t)), deleter};
```

Comment:
What is the fundamental difference between `tree->Clone` and `tree->GetDirectory()->CloneObject(tree)` that makes the latter ok but the first not?